### PR TITLE
Added power support for the travis.yml file with ppc64le. package: assertion-error

### PR DESCRIPTION
**Pull request by sreekanth370**
Added power support for the travis.yml file with ppc64le. This is part of the Ubuntu distribution for ppc64le. 
This helps us simplify testing later when distributions are re-building and re-releasing.

updated versions  node_js:
  - '10'
  - '12'
  - '14'

### CODE REVIEW

**Comment by sreekanth370** [comment]
Hi ,  kindly respond and merge the code with power architecture.

**Comment by keithamus** [comment]
Hello @sreekanth370 thanks for the PR

Could you please explain to me why we need to support `ppc64le` architectures? Have you found some particular edge cases that warrant running tests using that architecture?

**Comment by sreekanth370** [comment]
Hi, 

Not sure how familiar you are with the Power architecture - primarily the bigger enterprise class systems from IBM for banking/insurance/retail/etc.etc but this ensures that your code and all dependencies are tested on the Power architecture. Debian/Ubuntu, and Fedora/RHEL/CentOS have ppc64le (Power) distributions and this code goes into those distributions. This ensures that top of tree is always validated on ppc64le using (free) Power servers at Travis CI.

**Comment by sreekanth370** [comment]
kindly  review and merge the code with power support architecture

**Comment by sreekanth370** [comment]
For more info tag @gerrith3

**Comment by keithamus** [comment]
I'm familiar with PowerPC architecture, I'm just struggling to grasp why we'd specifically test for it? For example we don't test on 32 bit architectures, or ARM architectures, despite them likely being more prevalent. JavaScript - being an interpreted language - is abstracted away from issues of CPU architecture, and so it seems to be of little value to run tests on it. 

On the other hand, committing to running tests on ppc may limit the choice we have in future, if we intend to move to a different CI provider. Once encoded this decisions can be difficult to undo, so I'd like to make sure we're doing it for empirical reasons that have concrete benefits.

Is this library broken on ppc today? Has it ever been?

**Comment by keithamus** [comment]
I'll close this for now. When there's suitable objective reasons for doing this we can re-open and merge.

**Comment by gerrith3** [comment]
Hey @keithamus fair questions.  First, I believe you already use travis on every patch, and it is building with node v 0.1 so it hasn't been updated in a while.  0.1 isn't available on ppc64le (too old) so @sreekanth370 removed the old version and updated to some more recent versions and tested to make sure it works.

To help understand part of why we are doing this, we have been asked to support all npms on ppc64le, and set up an automation harness to test them all.  For being arch independent, oddly some 20-25% failed to build.  The problem, telling a priori which ones are going to fail.  We are trying to update our harness to more automatically find the right dependencies, etc. but some of this comes to making sure top of tree stays functional even after testing/porting once.  As part of this (this is just a small piece of our bigger ecosystem work) we have added free ppc64le hardware to travis and are trying to "lock in" the goodness of working builds.  I've asked my team to come up with a list of failure reasons but they are all over the board.

Basically, if anyone asked me today if any particular tree works on ppc64le I have to download, build, resolve issues, rebuild, repeat OR we can look at the latest build badge and see that ppc64le passed.

We are trying to do this with the least possible impact on communities, i.e. any failures get reported directly to the person that "broke the build" when it happens, rather than finding out months or years later that something changed and broke stuff.

The point is that this would be good for users of ppc64le hardware and basically free for maintainers.

**Comment by keithamus** [comment]
The concerns I have with adding support is that while we're using Travis today, that might not always be the case - for example we've been moving repos over to GitHub Actions which to my knowledge doesn't support PPC. If we add support it becomes difficult to take it away. 

Similarly, if actual issues arise around PPC we don't have the bandwidth to fix these. None of the dev team have PPC devices to do any extensive debugging, so we'd have to rely on Travis. The problem here is that Travis only provides a certain amount of monthly minutes, which means we'll be limited as to how much we can actually debug these issues. 

It sounds like your team has the resources to test & investigate issues should they arise, so I'd encourage you to do that. The chai team doesn't have those same resources, and while Travis offers us a little it doesn't offer us enough.

**Comment by gerrith3** [comment]
@keithamus first, there is a lot of FUD around about Travis, but  you can get as many credits as you need as an open source developer.  And, those credits may not even be necessary on Power - we are working on that with Travis now.

GHA may support ppc64le in the future - we are looking into it now but we don't have a quick answer on that.

My team is actually struggling - 8 people to get 80,000 packages working and keep them working is a bit of a challenge.  We normally port on demand from user requests, and are usuallyl backed up years...  The more we can do with CIs, the better, hence this drive to get people using Travis for not just Intel builds but also ppc64le, s390x, arm, etc.

Also, even if you move to GHA (which has limited minutes) you can run two CIs if you need - one of intel, one for ppc64le.  Not elegant but hopefully they will converge in the future.
